ecoscope/analysis/ecograph.py

```diff
@@ -197,7 +197,10 @@ def _get_ecograph(self, trajectory_gdf, individual_name, radius, cutoff, tortuos
 
         for node in G.nodes():
             for key in attributes.keys():
-                G.nodes[node][key] = np.mean(G.nodes[node][key])
+                if len(G.nodes[node][key]) != 0:
```

Comment:
i think this would be simpler using https://numpy.org/doc/stable/reference/generated/numpy.nanmean.html

Reply:
You are right : Using np.nanmean prevents the code from checking for nan values in other functions. I changed it. However, we still to keep the if len(G.nodes[node][key]) != 0, because it's possible for a given node to have no parameter values at all. In that case, using np.mean or np.nanmean will throw the warning « computing mean of empty slice » before returning np.nan. In order to prevent that warning, we need to check the length of the node attribute values.